### PR TITLE
CoinMP: backport build fixes for musl-libc

### PR DESCRIPTION
**Pull request by yu-re-ka**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/CoinMP/default.nix [hunk 1]
```diff
@@ -9,6 +9,18 @@ stdenv.mkDerivation rec {
     sha256 = "13d3j1sdcjzpijp4qks3n0zibk649ac3hhv88hkk8ffxrc6gnn9l";
   };
 
+  patches = [
+    # backport build fixes for pkgsMusl.CoinMP
+    # https://github.com/coin-or/Clp/commit/b637e1d633425ae21ec041bf7f9e06f56b741de0
+    ./0001-change-more-reinterpret_cast-from-NULL-to-C-cast-see.patch
+    # https://github.com/coin-or/Clp/commit/e749fe6b11a9patches/OpenMP/g006d744af2ca2691220862e3a59
```
Comment (alyssais) [inline pkgs/development/libraries/CoinMP/default.nix:16]: Something seems to have gone wrong with some of these URLs.

File: pkgs/development/libraries/CoinMP/default.nix [hunk 2]
```diff
@@ -9,6 +9,23 @@ stdenv.mkDerivation rec {
...
   };
 
+  patches = [
+    # backport build fixes for pkgsMusl.CoinMP
+    # https://github.com/coin-or/Clp/commit/b637e1d633425ae21ec041bf7f9e06f56b741de0
+    ./0001-change-more-reinterpret_cast-from-NULL-to-C-cast-see.patch
+    # https://github.com/coin-or/Clp/commit/e749fe6b11a90006d744af2ca2691220862e3a59
+    ./0002-change-reinterpret_cast-of-NULL-to-C-style-case-fixe.patch
+    # https://github.com/coin-or/Cgl/commit/57d8c71cd50dc27a89eaeb4672499bca55f1fd72
+    (fetchpatch {
```
Comment (alyssais) [inline pkgs/development/libraries/CoinMP/default.nix:19]: Needs a fetchpatch input.

**Review by alyssais — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/CoinMP/default.nix [hunk 3]
```diff
@@ -9,6 +9,23 @@ stdenv.mkDerivation rec {
     sha256 = "13d3j1sdcjzpijp4qks3n0zibk649ac3hhv88hkk8ffxrc6gnn9l";
   };
 
+  patches = [
+    # backport build fixes for pkgsMusl.CoinMP
+    # https://github.com/coin-or/Clp/commit/b637e1d633425ae21ec041bf7f9e06f56b741de0
+    ./0001-change-more-reinterpret_cast-from-NULL-to-C-cast-see.patch
+    # https://github.com/coin-or/Clp/commit/e749fe6b11a90006d744af2ca2691220862e3a59
+    ./0002-change-reinterpret_cast-of-NULL-to-C-style-case-fixe.patch
+    # https://github.com/coin-or/Cgl/commit/57d8c71cd50dc27a89eaeb4672499bca55f1fd72
```
Comment (alyssais) [inline pkgs/development/libraries/CoinMP/default.nix:18]: I guess this doesn't need the comment any more.

**Review by fzakaria — COMMENTED** [review]
LGTM -- why not bump CoinMP instead though than backport patches?

**Review by alyssais — APPROVED** [review]

**Pull request merged**